grammar.js

```diff
@@ -157,7 +159,7 @@ module.exports = grammar({
         seq(
           field('parameters', alias($.parameters, $.method_parameters)),
           choice(
-            seq(optional($._terminator), $._body_statement),
+            seq(optional($._terminator), optional(field("method_body", $.body)), 'end'),
```

Comment:
I think we should use the field name `'body'` instead of including the type of the enclosing node. Same for `do_block` , `class`, `module`, `singleton_class`, etc. 

I can see that having `namespace_body` makes it easy to capture the bodies of all classes/modules/singleton classes. However, this causes problems for tools that generate data models based on the grammar. For example https://github.com/github/codeql/blob/main/ruby/ql/lib/codeql/ruby/ast/internal/TreeSitter.qll#L1104-L1124 would end up having things like`Method::getMethodBody()` and `DoBlock::getDoBlockBody()` which look unnecessarily repetitive.

It might be worth trying if defining a rule named `_namespace: $ => choice($.module, $.class, $.singleton_class)` and adding `_namespace` to the `supertypes` list. I don't know much about the queries you are writing, but hopefully the query syntax allows selecting the `body` field of all nodes of type `_namespace`.